packages/zoe/src/contracts/loan/types.js

```diff
@@ -0,0 +1,183 @@
+ *
+ * The loan now has a lenderSeat, which is added to the config.
+ */
+
+/**
+ * @typedef BorrowerConfigProperties
+ *
+ * @property {ZCFSeat} collateralSeat
+ *
+ *   The ZCFSeat holding the collateral in escrow after the borrower
+ *   escrows it
+ *
+ * @property {() => Amount} getDebt
+ *
+ *   A function to get the current debt
+ *
+ * @property {PromiseKit} liquidationPromiseKit
+ *
+ *   PromiseKit that includes a promise that resolves to a PriceQuote
+ *   when liquidation is triggered
+ */
+
+/**
+ * @typedef {LoanConfigWithLender & BorrowerConfigProperties } LoanConfigWithBorrower
+ *
+ * The loan has a lender, a borrower, and collateral escrowed.
+ */
+
+/**
+ * @callback ScheduleLiquidation
+ * @param {ContractFacet} zcf
+ * @param {LoanConfigWithBorrower} config
+ */
+
+/**
+ * @callback MakeLendInvitation
+ * @param {ContractFacet} zcf
+ * @param {LoanTerms} config
+ * @returns {Promise<Invitation>} lendInvitation
+ */
+
+/**
+ * @callback MakeBorrowInvitation
+ * @param {ContractFacet} zcf
+ * @param {LoanConfigWithLender} config
+ * @returns {Promise<Invitation>} borrowInvitation
+ */
+
+/**
+ * @callback MakeCloseLoanInvitation
+ * @param {ContractFacet} zcf
+ * @param {LoanConfigWithBorrower} config
+ * @returns {Promise<Invitation>} closeLoanInvitation
+ */
+
+/**
+ * Allows holder to add collateral to the contract. Exits the seat
+ * after adding.
+ *
+ * @callback MakeAddCollateralInvitation
+ * @param {ContractFacet} zcf
+ * @param {LoanConfigWithBorrower} config
+ * @returns {Promise<Invitation>} addCollateralInvitation
+ */
+
+/**
+ * @callback Liquidate
+ * @param {ContractFacet} zcf
+ * @param {LoanConfigWithBorrower} config
+ * @returns {void}
+ */
+
+/**
+ * @callback MakeDebtCalculator
+ * @param {DebtCalculatorConfig} debtCalculatorConfig
+ */
+
+/**
+ * @callback CalcInterestFn
+ * @param {number} oldDebtValue
+ * @param {number} interestRate
+ * @returns {number} interest
+ */
+
+/**
+ * @typedef {Object} DebtCalculatorConfig
+ * @property {CalcInterestFn} calcInterestFn
+ *
+ *   A function to calculate the interest, given the debt value and an
+ *   interest rate in basis points.
+ *
+ * @property {Amount} originalDebt
+ *
+ *   The debt at the start of the loan, in Loan brand
+ *
+ * @property {AmountMath} loanMath
+ *
+ *   AmountMath for the loan brand
+ *
+ * @property {PeriodAsyncIterable} periodAsyncIterable
+ *
+ *   The AsyncIterable to notify when a period has occurred
+ *
+ * @property {number} interestRate
+ */
+
+/**
+ * @typedef {Object} BorrowFacet
+ *
+ * @property {() => Promise<Invitation>} makeCloseLoanInvitation
+ *
+ * Make an invitation to close the loan by repaying the debt
+ *   (including interest).
+ *
+ * @property {() => Promise<Invitation>} makeAddCollateralInvitation
+ *
+ * Make an invitation to add collateral to protect against liquidation
+ *
+ * @property {() => Promise<PriceQuote>} getLiquidationPromise
+ *
+ * Get a promise that will resolve if liquidation occurs
```

Comment:
If it's a Promise for a priceQuote, what does the quote represent?

Reply:
It's the quote given by the priceAuthority that caused the liquidation to happen. So it's the value of the collateral at the time it was below the mmr. If the priceAuthority is slow, this could be very different than the level that was registered as the trigger amount